guides/source/asset_pipeline.md

```diff
@@ -100,194 +99,278 @@ content into the name, usually at the end. For example a CSS file `global.css`
+Pinning "scheduler" to https://ga.jspm.io/npm:scheduler@0.20.2/index.js
 
-### Asset Organization
+./bin/importmap json
 
-Pipeline assets can be placed inside an application in one of three locations:
-`app/assets`, `lib/assets` or `vendor/assets`.
-
-* `app/assets` is for assets that are owned by the application, such as custom
-images, JavaScript files, or stylesheets.
+{
+  "imports": {
+    "application": "/assets/application-37f365cbecf1fa2810a8303f4b6571676fa1f9c56c248528bc14ddb857531b95.js",
+    "react": "https://ga.jspm.io/npm:react@17.0.2/index.js",
+    "react-dom": "https://ga.jspm.io/npm:react-dom@17.0.2/index.js",
+    "object-assign": "https://ga.jspm.io/npm:object-assign@4.1.1/index.js",
+    "scheduler": "https://ga.jspm.io/npm:scheduler@0.20.2/index.js"
+  }
+}
+```
 
-* `lib/assets` is for your own libraries' code that doesn't really fit into the
-scope of the application or those libraries which are shared across applications.
+As you can see, the two packages react and react-dom resolve to a total of four dependencies, when resolved via the jspm default.
 
-* `vendor/assets` is for assets that are owned by outside entities, such as
-code for JavaScript plugins and CSS frameworks. Keep in mind that third party
-code with references to other files also processed by the asset Pipeline (images,
-stylesheets, etc.), will need to be rewritten to use helpers like `asset_path`.
+Now you can use these in your `application.js` entrypoint like you would any other module:
 
-#### Search Paths
+```javascript
+import React from "react"
+import ReactDOM from "react-dom"
+```
 
-When a file is referenced from a manifest or a helper, Sprockets searches the
-three default asset locations for it.
+You can also designate a specific version to pin:
 
-The default locations are: the `images`, `javascripts` and `stylesheets`
-directories under the `app/assets` folder, but these subdirectories
-are not special - any path under `assets/*` will be searched.
+```sh
+./bin/importmap pin react@17.0.1
+Pinning "react" to https://ga.jspm.io/npm:react@17.0.1/index.js
+Pinning "object-assign" to https://ga.jspm.io/npm:object-assign@4.1.1/index.js
+```
 
-For example, these files:
+Or even remove pins:
 
+```sh
+./bin/importmap unpin react
+Unpinning "react"
+Unpinning "object-assign"
 ```
-app/assets/javascripts/home.js
-lib/assets/javascripts/moovinator.js
-vendor/assets/javascripts/slider.js
-vendor/assets/somepackage/phonebox.js
+
+You can control the environment of the package for packages with separate "production" (the default) and "development" builds:
+
+```sh
+./bin/importmap pin react --env development
+Pinning "react" to https://ga.jspm.io/npm:react@17.0.2/dev.index.js
+Pinning "object-assign" to https://ga.jspm.io/npm:object-assign@4.1.1/index.js
 ```
 
-would be referenced in a manifest like this:
+You can also pick an alternative, supported CDN provider when pinning, like `unpkg` or `jsdelivr` (`jspm` is the default):
```

Comment:
Just for convenience: 
`unpkg` -> [`unpkg`](https://unpkg.com/)
`jsdelivr` -> [`jsdelivr`](https://www.jsdelivr.com/)